scripts/deploy/src/install.js

```diff
@@ -0,0 +1,136 @@
+  if (!namespace) {
+    console.error('Namespace is not specified.');
+    process.exit(1);
+  }
+  return namespace;
+};
+
+const get_image_tag = async function(chtversion) {
+  const response = await fetch(`https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:${chtversion}`);
+  const data = await response.json();
+  const tag = data.tags && data.tags.find(tag => tag.container_name === 'cht-api');
+  if (!tag) {
+    return Promise.reject(new UserRuntimeError('cht image tag not found'));
+  }
+  return tag.image.split(':').pop();
+};
+
+const get_chart_version = function(values) {
+  return values.cht_chart_version || DEFAULT_CHART_VERSION;
+};
+
+const helmInstallOrUpdate = function(valuesFile, namespace, values, image_tag) { //NoSONAR
+  const chart_version = get_chart_version(values);
+  ensureMedicHelmRepo();
+  const project_name = values.project_name || '';
+  const namespaceExists = checkNamespaceExists(namespace);
+
+  try {
+    const releaseExists = child_process.execSync(`helm list -n ${namespace}`).toString(); //NoSONAR
+    if (releaseExists.includes(project_name)) { //NoSONAR
+      console.log('Release exists. Performing upgrade.');
+      child_process.execSync(`helm upgrade` + //NoSONAR
+        ` --install ${project_name} ${CHT_CHART_NAME}` + //NoSONAR
+        ` --version ${chart_version} --namespace ${namespace}` + //NoSONAR
+        ` --values ${valuesFile} --set cht_image_tag=${image_tag}`, { stdio: 'inherit' }); //NoSONAR
+      console.log(`Instance at ${values.ingress.host} upgraded successfully.`);
+    } else {
+      console.log('Release does not exist. Performing install.');
+      const createNamespaceFlag = namespaceExists ? '' : '--create-namespace';
+      child_process.execSync(`helm install ${project_name} ${CHT_CHART_NAME}` + //NoSONAR
+        ` --version ${chart_version} --namespace ${namespace} ${createNamespaceFlag}` + //NoSONAR
+        ` --values ${valuesFile} --set cht_image_tag=${image_tag}`, { stdio: 'inherit' }); //NoSONAR
+      console.log(`Instance at ${values.ingress.host} installed successfully.`);
+    }
+  } catch (err) {
+    console.error(JSON.stringify(err));
+    process.exit(1);
+  }
+};
+
+const checkNamespaceExists = function(namespace) {
+  try {
+    const result = child_process.execSync(`kubectl get namespace ${namespace}`).toString(); //NoSONAR
+    return result.includes(namespace); //NoSONAR
+  } catch (err) {
+    return false;
+  }
+};
+
+const ensureMedicHelmRepo = function() {
+  try {
+    const repoList = child_process.execSync(`helm repo list -o json`).toString();
+    const repos = JSON.parse(repoList);
+    const medicRepo = repos.find(repo => repo.name === MEDIC_REPO_NAME);
+    if (!medicRepo) {
+      console.log(`Helm repo ${MEDIC_REPO_NAME} not found, adding..`);
+      child_process.execSync(`helm repo add ${MEDIC_REPO_NAME} ${MEDIC_REPO_URL}`, { stdio: 'inherit' }); //NoSONAR
+    } else if (medicRepo.url.replace(/\/$/, '') !== MEDIC_REPO_URL) {
```

Comment:
So ... the expected correct value is `${MEDIC_REPO_URL}/` ? I don't understand why we would be removing a trailing slash in this conditional.

Reply:
It's to account for helm behavior where the trailing slash exists in some versions and not in others.